### PR TITLE
Feature/erssup 56909

### PR DESCRIPTION
**Pull request by nhsd-jack-wainwright**
## Summary
* Routine Change
* :exclamation: Breaking Change
* :robot: Operational or Infrastructure Change
* :sparkles: New Feature
* :warning: Potential issues that might be caused by this change

Add any other relevant notes or explanations here. **Remove this line if you have nothing to add.**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: specification/components/schemas/eRS-ChangeShortlist-Parameters.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+          type: string
+          enum:
+            - "https://fhir.nhs.uk/STU3/StructureDefinition/eRS-ChangeShortlist-Parameters-1"
+          example: "https://fhir.nhs.uk/STU3/StructureDefinition/eRS-ChangeShortlist-Parameters-1"
+  parameter:
+    type: array
+    description: |
+      | Field                             | Cardinality | Notes                                                                                    |
+      | --------------------------------- | ----------- | ---------------------------------------------------------------------------------------- |
+      | `Shortlist`                       | `1..1`      | Shortlist indicating the new services a referral should have shortlisted                 |                                           |
```
Comment (nhsd-hubert-raich) [inline specification/components/schemas/eRS-ChangeShortlist-Parameters.yaml:31]: Extra pipe at the end of this line, but somehow preview renders the table correctly.

File: specification/e-referrals-service-api.yaml [hunk 2]
```diff
@@ -1292,6 +1292,39 @@ paths:
...
+          application/fhir+json:
+            schema:
+              $ref: 'components/schemas/eRS-ChangeShortlist-Parameters.yaml'
+            examples:
+              unbookedReferralRequest:
+                summary: Unbooked ReferralRequest
+                description: Changing the service list of an unbooked `ReferralRequest`
+                value:
+                  $ref: 'components/examples/changeShortlist/requests/UnbookedReferral.json'
+      responses:
```
Comment (nhsd-hubert-raich) [inline specification/e-referrals-service-api.yaml:1324]: Shoudn't we document some error responses here? At least the once common to all endpoints ?

**Review by nhsd-hubert-raich — CHANGES_REQUESTED** [review]

File: specification/e-referrals-service-api.yaml (hunk 2 above)
Comment (nhsd-jack-wainwright) [reply specification/e-referrals-service-api.yaml:1324]: We should yup, have added in the error codes now :)

File: specification/components/schemas/eRS-ChangeShortlist-Parameters.yaml (hunk 1 above)
Comment (nhsd-jack-wainwright) [reply specification/components/schemas/eRS-ChangeShortlist-Parameters.yaml:31]: Hmm not sure how this is rendering correctly, removed this now.

File: sandbox/src/mocks/changeShortlist/requests/UnbookedReferral.json [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+                            "identifier": {
+                                "system": "http://fhir.nhs.net/Id/ers-service",
+                                "value": "11002"
+                            }
+                        }
+                    }
+                ],
+                "extension": [
+                    {
+                        "url": "https://fhir.nhs.uk/STU3/StructureDefinition/Extension-eRS-Shortlist-SearchCriteria-1",
```
Comment (adamoldfield-nhs) [inline sandbox/src/mocks/changeShortlist/requests/UnbookedReferral.json:77]: Looks like this extension is missing from the schema. Missing for create referral too I suspect.

Comment (nhsd-jack-wainwright) [reply sandbox/src/mocks/changeShortlist/requests/UnbookedReferral.json:77]: Looks to be yup, added this extension to the OAS component for the shortlist now.

**Review by adamoldfield-nhs — CHANGES_REQUESTED** [review]
I focussed my review on the generated documentation I didn't look in detail at the files changed here. I'll leave that to Hubert.

Just one comment about the missing extension for the search criteria.

**Review by nhsd-hubert-raich — APPROVED** [review]

**Review by adamoldfield-nhs — APPROVED** [review]
Change to add search criteria extension to the shortlist looks good.

**Pull request merged**